File: Java/169.多数元素.java
/*
 * @lc app=leetcode.cn id=169 lang=java
 *
 * [169] 多数元素
 */

// @lc code=start
class Solution {
    public int majorityElement(int[] nums) {
        int candidate = -1, count = 0;

        for(int num : nums){
            if(count == 0)
                candidate = num;
            
            count += (num == candidate) ? 1 : -1;
        }

        return candidate;
    }
}
// @lc code=end

File: Python/169.多数元素.py
#
# @lc app=leetcode.cn id=169 lang=python3
#
# [169] 多数元素
#
from typing import List
# @lc code=start
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        candidate = -1
        count = 0

        for num in nums:
            if count == 0:
                candidate = num

            if candidate == num:
                count += 1
            else:
                count -= 1

        return candidate
# @lc code=end

